### PR TITLE
[TorchAcc] Update to TorchAcc2.3 and transformers 4.40.2

### PR DESCRIPTION
**Pull request by baoleai**
# PR type
- [ ] Bug Fix
- [x] New Feature
- [ ] Document Updates
- [ ] More Models or Datasets Support

# PR information
Revised the torchacc-related interfaces to accommodate torchacc 2.3 and transformers 4.40.2; added support for DDP's save and load checkpoint functionality.

### CODE REVIEW

File: swift/utils/torch_utils.py [hunk 1]
```diff
@@ -75,11 +75,11 @@ def is_local_master():
 
 
 def use_torchacc() -> bool:
-    return os.getenv('USE_TORCHACC', '0') == '1'
+    return os.getenv('USE_TORCHACC', '0').upper() in ['ON', '1', 'YES', 'TRUE', 'Y']
```
Comment (Jintao-Huang) [inline swift/utils/torch_utils.py]: 有个函数是这个功能 from transformers.utils import strtobool
不过不改也没关系

File: swift/llm/tuner.py [hunk 2]
```diff
@@ -224,9 +224,9 @@ def prepare_model(model, args: SftArguments):
                 model = Swift.prepare_model(model, boft_config)
                 logger.info(f'boft_config: {boft_config}')
         else:
-            if use_torchacc():
+            if use_torchacc() and args.fsdp_num > 1:
                 consolidate_checkpoint(args.resume_from_checkpoint, 'adapter_model')
-            model = Swift.from_pretrained(model, args.resume_from_checkpoint, is_trainable=True)
+            model = Swift.from_pretrained(model, args.resume_from_checkpoint, adapter_name='default', is_trainable=True)
```
Comment (Jintao-Huang) [inline swift/llm/tuner.py]: 这个地方 @tastelikefeet 看一下

File: swift/utils/torch_utils.py (hunk 1 above)
Comment (baoleai) [reply swift/utils/torch_utils.py]: fixed.

File: swift/llm/tuner.py [hunk 3]
```diff
@@ -224,9 +224,9 @@ def prepare_model(model, args: SftArguments):
                 model = Swift.prepare_model(model, boft_config)
                 logger.info(f'boft_config: {boft_config}')
         else:
-            if use_torchacc():
+            if use_torchacc() and args.fsdp_num > 1:
```
Comment (Zhikaiiii) [inline swift/llm/tuner.py]: 这里如果导入的时候不是torchacc格式的ckpt会有问题不？

File: swift/llm/tuner.py (hunk 2 above)
Comment (baoleai) [reply swift/llm/tuner.py]: fixed.

File: swift/llm/tuner.py (hunk 3 above)
Comment (baoleai) [reply swift/llm/tuner.py]: 不用torchacc时，保持了原行为

**Review by tastelikefeet — APPROVED** [review]

**Pull request merged**